server/src-lib/Hasura/GraphQL/Transport/WebSocket/Protocol.hs

```diff
@@ -128,11 +129,14 @@ encodeServerMsg msg =
     , ("payload", encJFromJValue connErr)
     ]
 
-  SMData (DataMsg opId payload) ->
-    [ encTy SMT_GQL_DATA
-    , ("id", encJFromJValue opId)
-    , ("payload", encodeGraphqlResponse payload)
-    ]
+  SMData (DataMsg opId payload execTime) ->
+    let encPayload = encodeGraphqlResponse payload
+    in [ encTy SMT_GQL_DATA
+       , ("id", encJFromJValue opId)
+       , ("response_size", encJFromJValue $ BL.length $ encJToLBS encPayload)
+       , ("query_execution_time", encJFromJValue execTime)
+       , ("payload", encPayload)
+       ]
```

Comment:
Just for my understanding, looks like we are sending the query execution time, response size etc. to the client in response as well?

Reply:
Yeah. Forgot this. However, the internal payload is sent in bytestring already. We have to decode JSON again. One solution I think that we don't need to include `response_size` on HGE. Lux will parse and caculate response size

About `query_execution_time`, I will move it outside

Reply:
Those details should only be logged right? Nothing in the responses should change?

Reply:
> he internal payload is sent in bytestring already. We have to decode JSON again to get response size.

We can just measure the length of the bytestring? That's how it is done in the HTTP layer currently.

Reply:
yes, just for logs only. 
> We can just measure the length of the bytestring? That's how it is done in the HTTP layer currently.

The bytestring response include more data `type` and `id` beside payload. If we accept them in `response_size`, it will be ok.

Reply:
I think that's okay (to include the entire apollo-protocol frame to count the response size). Because the `response_size` metric indicates how many bytes the server is sending out for each response. Thoughts @shahidhk ?

Reply:
Yeah, it can be the entire frame size.